test/unit/materialized_view_r3_emission_test.sql

```diff
@@ -0,0 +1,142 @@
+set client_encoding = 'utf-8';
+set client_min_messages = warning;
+create extension if not exists pgtap;
+reset client_min_messages;
+
+begin;
+
+select plan(9);
+
+select has_materialized_view(
+    'swrs_transform', 'r3_emission',
+    'swrs_transform.r3_emission should be a materialized view'
+);
+
+select has_index(
+    'swrs_transform', 'r3_emission', 'ggircs_r3_emission_primary_key',
+    'swrs_transform.r3_emission should have a primary key'
+);
+
+select columns_are('swrs_transform'::name, 'r3_emission'::name, array[
+    'id'::name,
+    'eccc_xml_file_id'::name,
+    'activity_name'::name,
+    'emissions_idx'::name,
+    'emission_idx'::name,
+    'gas_type'::name,
+    'quantity'::name,
+    'calculated_quantity'::name,
+    'cas_number'::name
+]);
+
+-- Insert data for fixture based testing
+insert into swrs_extract.eccc_xml_file (xml_file) values ($$
+<ReportData xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+  <ReportDetails>
+    <ReportType>R3</ReportType>
+  </ReportDetails>
+  <ActivityData xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <TotalAttributableEmissions>
+      <TotalGroups TotalGroupType="TotalAttribEmissions">
+        <Totals>
+          <Emissions EmissionsGasType="AttributableEmissions">
+            <TotalRow>
+              <Quantity>7437.645</Quantity>
+              <CalculatedQuantity>7437.645</CalculatedQuantity>
+              <CasNumber> 124-38-9</CasNumber>
```

Comment:
Is the space before ` 124-38-9` present in the production data, and in the extracted value? If so, we might want to trim whitespaces

Reply:
It's not in the prod data, I added it to check that we were properly removing whitespace with normalize-space()

Reply:
Great! Can you add a comment above then, so that someone like me doesn't think this is an accidental whitespace and removes it :smile: ?